File: 0881-boats-to-save-people/0881-boats-to-save-people.py
class Solution(object):
    def numRescueBoats(self, people, limit):
        people.sort()
        light=0
        heavy=len(people)-1
        boat=0
        while light<=heavy:
            if people[light]+people[heavy]<=limit:
                light+=1
            heavy-=1
            boat+=1
        return boat

        

File: 0881-boats-to-save-people/0881-boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        sort(people.begin(),people.end());
        int light=0,heavy=people.size()-1;
        int boat=0;
        while(light<=heavy){
            if (people[light]+people[heavy]<=limit){
                light++;
            }
            heavy--;
            boat++;
        }
        return boat;
    }
};